app/src/main/java/org/oppia/app/player/state/InteractionAdapter.kt

```diff
@@ -0,0 +1,202 @@
+import kotlinx.android.synthetic.main.item_selection_interaction_items.view.item_selection_checkbox
+import kotlinx.android.synthetic.main.item_selection_interaction_items.view.item_selection_contents_text_view
+import kotlinx.android.synthetic.main.multiple_choice_interaction_items.view.multiple_choice_content_text_view
+import kotlinx.android.synthetic.main.multiple_choice_interaction_items.view.multiple_choice_radio_button
+import org.oppia.app.R
+import org.oppia.app.databinding.ItemSelectionInteractionItemsBinding
+import org.oppia.app.databinding.MultipleChoiceInteractionItemsBinding
+import org.oppia.app.model.InteractionObject
+import org.oppia.app.model.StringList
+import org.oppia.app.player.state.itemviewmodel.SelectionInteractionContentViewModel
+import org.oppia.app.player.state.itemviewmodel.SelectionInteractionCustomizationArgsViewModel
+import org.oppia.app.player.state.listener.ItemClickListener
+import org.oppia.util.parser.HtmlParser
+
+private const val VIEW_TYPE_RADIO_BUTTONS = 1
+private const val VIEW_TYPE_CHECKBOXES = 2
+private const val INTERACTION_ADAPTER_TAG = "Interaction Adapter"
+
+/**
+ * Adapter to bind the interactions to the [RecyclerView]. It handles MultipleChoiceInput
+ * and ItemSelectionInput interaction views.
+ * */
+class InteractionAdapter(
+  private val htmlParserFactory: HtmlParser.Factory,
+  private val entityType: String,
+  private val explorationId: String,
+  private val itemList: MutableList<SelectionInteractionContentViewModel>,
+  private val selectionInteractionCustomizationArgsViewModel: SelectionInteractionCustomizationArgsViewModel,
+  private val itemClickListener: ItemClickListener,
+  private val selectedInputItemIndexes: ArrayList<Int>,
+  private val selectInputItemsListener: SelectInputItemsListener
+) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+
+  private var itemSelectedPosition = -1
+  private var selectedAnswerIndex = -1
+  private var selectedHtmlStringList = mutableListOf<String>()
+
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+    return when (viewType) {
+      VIEW_TYPE_RADIO_BUTTONS -> {
+        val inflater = LayoutInflater.from(parent.context)
+        val binding =
+          DataBindingUtil.inflate<MultipleChoiceInteractionItemsBinding>(
+            inflater,
+            R.layout.multiple_choice_interaction_items,
+            parent,
+            /* attachToParent= */ false
+          )
+        MultipleChoiceViewHolder(binding)
+      }
+      VIEW_TYPE_CHECKBOXES -> {
+        val inflater = LayoutInflater.from(parent.context)
+        val binding =
+          DataBindingUtil.inflate<ItemSelectionInteractionItemsBinding>(
+            inflater,
+            R.layout.item_selection_interaction_items,
+            parent,
+            /* attachToParent= */ false
+          )
+        ItemSelectionViewHolder(binding)
+      }
+      else -> throw IllegalArgumentException("Invalid view type")
```

Comment:
Update this with `else -> throw IllegalArgumentException("Invalid view type: $viewType")` for better error

Reply:
Done.